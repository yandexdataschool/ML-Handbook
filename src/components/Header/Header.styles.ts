import styled from 'styled-components';
import backVectorImage from './images/Vector.svg';
import findImage from './images/Find.svg';
import tumblerImage from './images/Tumbler.svg';

const Header = styled.header`
  margin-top: 0;
  margin-bottom: 0;
  max-width: 1440px;
  height: 52px;
  display: flex;
  justify-content: space-around;
`;

const Title = styled.h1`
  display: none;
  @media screen and (min-width: 1400px) {
    display: flex;
    width: 720px;
    height: 24px;
    margin-block: 0;
    margin: auto 0;
    margin-left: 40px;
    font-family: 'Yandex Sans Text';
    font-weight: 300;
    font-style: normal;
    font-size: 20px;
  }
`;

const BackToMain = styled.div`
  width: 280px;
  height: 24px;
  margin: auto 0;
  display: flex;
`;

const BackVector = styled.div`
  width: 14px;
  height: 10px;
  margin: auto 0;
  background-repeat: no-repeat;
  background-image: url(${backVectorImage});
`;

const BackTitle = styled.h2`
  width: 250px;
  height: 24px;
  margin-left: 6px;
  margin-block: 0;
  font-family: 'Yandex Sans Text';
  font-weight: 500;
  font-style: normal;
  font-size: 20px;
  color: rgba(59, 72, 86, 1);
`;

const WBmode = styled.div`
  display: none;
  @media screen and (min-width: 768px) {
    margin: auto 0;
    margin-left: 10px;
    width: 204px;
    height: 21px;
    font-family: 'Yandex Sans Text';
    font-weight: 500;
    font-style: normal;
    font-size: 14px;
    color: rgba(59, 72, 86, 1);
    display: flex;
  }
`;

const Theme = styled.div`
  width: 98px;
  display: flex;
`;

const Tumbler = styled.div`
  width: 46px;
  height: 16px;
  padding-left: 52px;
  background-image: url(${tumblerImage});
  background-repeat: no-repeat;
`;

const Find = styled.div`
  width: 32px;
  height: 32px;
  margin: auto 0;
  background-image: url(${findImage});
`;

export { Header, Title, BackToMain, BackVector, BackTitle, WBmode, Theme, Tumbler, Find };
