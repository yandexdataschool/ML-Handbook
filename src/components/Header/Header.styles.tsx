import styled from 'styled-components';

const HeaderStyled = styled.div`
  margin-top: 0;
  margin-bottom: 0;
  width: 1440px;
  height: 52px;
  display: flex;
`;

const Title = styled.h1`
  width: 720px;
  height: 24px;
  font-family: 'Yandex Sans Text';
  font-weight: 300;
  font-style: normal;
  font-size: 20px;
  line-height: 25px;
  margin-left: 10px;
`;

const Back = styled.h2`
  width: 250px;
  height: 24px;
  margin-left: 50px;
  font-family: 'Yandex Sans Text';
  font-weight: 500;
  font-style: normal;
  font-size: 20px;
  line-height: 24px;
  color: rgba(59, 72, 86, 1);
`;

const WBmode = styled.h2`
  width: 236px;
  height: 21px;
  margin-left: 28px;
  font-family: 'Yandex Sans Text';
  font-weight: 500;
  font-style: normal;
  font-size: 14px;
  line-height: 17.5px;
  color: rgba(59, 72, 86, 1);
`;

const Find = styled.div`
  width: 32px;
  height: 32px;
  margin-left: 54px;
`;

export { HeaderStyled, Title, Back, WBmode, Find };
