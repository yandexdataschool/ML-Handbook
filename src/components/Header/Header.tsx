import * as React from 'react'

const headingStyles = {
  marginTop: 0,
  marginBottom: 64,
  maxWidth: 320,
}
const headingAccentStyles = {
  color: "#663399",
}

export const Header = () => {
  return (
    <h1 style={headingStyles}>
      Congratulations
      <br />
      <span style={headingAccentStyles}>— you just made a Gatsby site! </span>
      <span role="img" aria-label="Party popper emojis">
          🎉🎉🎉
        </span>
    </h1>
  )
}
