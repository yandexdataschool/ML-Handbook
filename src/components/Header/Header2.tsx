import React from 'react';
import { HeaderStyled, Title, BackToMain, BackVector, BackTitle, WBmode, Find } from './Header.styles';

export default function Header() {
  return (
    <HeaderStyled>
      <BackToMain>
        <BackVector />
        <BackTitle>ML-учебник</BackTitle>
      </BackToMain>
      <Title>Вероятностный подход в ML</Title>
      <WBmode />
      <Find />
    </HeaderStyled>
  );
}
