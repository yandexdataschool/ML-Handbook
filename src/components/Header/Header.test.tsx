import * as React from "react"
import * as renderer from "react-test-renderer"

import { Header } from './Header'

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Header />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
