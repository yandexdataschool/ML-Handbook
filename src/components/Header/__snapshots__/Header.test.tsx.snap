// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<h1
  style={
    Object {
      "marginBottom": 64,
      "marginTop": 0,
      "maxWidth": 320,
    }
  }
>
  Congratulations
  <br />
  <span
    style={
      Object {
        "color": "#663399",
      }
    }
  >
    — you just made a Gatsby site! 
  </span>
  <span
    aria-label="Party popper emojis"
    role="img"
  >
    🎉🎉🎉
  </span>
</h1>
`;
